ojdbc-provider-jackson-oson/src/main/java/oracle/jdbc/provider/oson/OsonParser.java

```diff
@@ -0,0 +1,703 @@
+
+      case START_ARRAY:
+        return JsonToken.START_ARRAY;
+      case END_ARRAY:
+        return JsonToken.END_ARRAY;
+
+      case KEY_NAME:
+        if(DEBUG) logger.log(Level.FINEST, "KEY_NAME> " + parser.getString());
+        this.fieldName = parser.getString();
+        return JsonToken.FIELD_NAME;
+
+      case VALUE_STRING:
+        if(DEBUG) logger.log(Level.FINEST, "VALUE_STRING> " + parser.getString());
+        return JsonToken.VALUE_STRING;
+
+      case VALUE_DECIMAL:
+        if(DEBUG) logger.log(Level.FINEST, "VALUE_DECIMAL> " + parser.getBigDecimal()+" / "+parser.isIntegralNumber());
+        return parser.isIntegralNumber() ? JsonToken.VALUE_NUMBER_INT : JsonToken.VALUE_NUMBER_FLOAT;
+
+      case VALUE_FLOAT:
+        if(DEBUG) logger.log(Level.FINEST, "VALUE_FLOAT> " + parser.getFloat());
+        return JsonToken.VALUE_NUMBER_FLOAT;
+
+      case VALUE_DOUBLE:
+        if(DEBUG) logger.log(Level.FINEST, "VALUE_DOUBLE> " + parser.getDouble());
+        return JsonToken.VALUE_NUMBER_FLOAT;
+
+      case VALUE_NULL:
+        if(DEBUG) logger.log(Level.FINEST, "VALUE_NULL");
+        return JsonToken.VALUE_NULL;
+
+      case VALUE_TRUE:
+        if(DEBUG) logger.log(Level.FINEST, "VALUE_TRUE");
+        return JsonToken.VALUE_TRUE;
+
+      case VALUE_FALSE:
+        if(DEBUG) logger.log(Level.FINEST, "VALUE_FALSE");
+        return JsonToken.VALUE_FALSE;
+
+      case VALUE_TIMESTAMP:
+        if(DEBUG) logger.log(Level.FINEST, "VALUE_TIMESTAMP");
+        return JsonToken.VALUE_STRING;
+
+      case VALUE_TIMESTAMPTZ:
+        if(DEBUG) logger.log(Level.FINEST, "VALUE_TIMESTAMPTZ");
+        return JsonToken.VALUE_STRING;
+        
+      case VALUE_BINARY:
+        if(DEBUG) logger.log(Level.FINEST, "VALUE_BINARY");
+        return JsonToken.VALUE_STRING;
+        
+      case VALUE_INTERVALDS:
+        if(DEBUG) logger.log(Level.FINEST, "VALUE_INTERVALDS");
+        return JsonToken.VALUE_STRING;
+        
+      case VALUE_INTERVALYM:
+        if(DEBUG) logger.log(Level.FINEST, "VALUE_INTERVALYM");
+        return JsonToken.VALUE_STRING;
+
+      default:
+        throw new IllegalStateException("event: " + currentEvent);
+    }
+
+  }
+
+  /**
+   * Get the current token.
+   * @return
+   */
+  @Override
+  public JsonToken currentToken() {
+    return getCurrentToken();
+  }
+
+  /**
+   * Get the current Token ID.
+   */
+  @Override
+  public int getCurrentTokenId() {
+    if(DEBUG) logger.log(Level.FINEST, "getCurrentTokenId");
+    JsonToken jt;
+    if ((jt = OSON_EVENT_TO_JSON_TOKEN.get(currentEvent)) != null) {
+      return jt.id();
+    } else if( currentEvent == OracleJsonParser.Event.VALUE_DECIMAL ) {
+      return parser.isIntegralNumber() ? JsonToken.VALUE_NUMBER_INT.id() : JsonToken.VALUE_NUMBER_FLOAT.id();
+    }
+
+    return -1;
+  }
+
+  /**
+   * Get the current Token ID.
+   */
+  @Override
+  public int currentTokenId() {
+    if(DEBUG) logger.log(Level.FINEST, "getCurrentTokenId");
```

Comment:
Looks very similar to the method above? can getCurrentTokenId be called instead?

Reply:
done